### DISCUSSION TITLE
Mineflayer not working on replit

### DISCUSSION THREAD

**Discussion by Miloldr** [PrismarineJS/mineflayer#3278] [Q&A]
Same code that works perfectly fine on my computer, does not on replit platform. Now there is few weird things going on, **sometimes** mineflayer works but for once, then it doesn't for a very long time and just gives etimedout error after a long while and does create bot object but doesn't connect to the server and the object has things like bot.player undefined. Also it worked few months ago I'm pretty sure and back then it wouldn't work and bot doesn't connect but "kill 1" to the console and restarting the program worked (probably because it clears the memory or / and changes IP). The code? Anything that works on computer.

**Comment by ajiegenrald** [comment]
Maybe is your code problem, its working fine on me. , Replit kills free hosting anyway, you can use other platform.

**Reply by Miloldr** [reply]
I don't want to spend money :>

**Reply by Mipppy** [reply]
You should just use VS Code

**Comment by Mipppy** [comment]
Mineflayer does indeed work on relit.com, I've hosted many bots from it. The issue is that running it, installing packages, and setting up environment, take a while. All Mineflayer plugins also work as intended

**Reply by Miloldr** [reply]
When. Same code indeed worked year ago, so when.

**Reply by Mipppy** [reply]
I ran a bot off it yesterday.  It's actually quite useful if because it doubles as a VPN if you are willing to wait for loads

**Reply by Miloldr** [reply]
Doesn't work for me NOW but it did quite a while before, there was some problem like bot not starting and timeouting which I think I fixed by kill 1 in shell and running again. Now same code doesn't work. If you mind, can you give me those first lines of code of initialization then. I guess it can be the "year old code"

**Reply by Mipppy** [reply]
I'm not on my computer now, but I will once I am.  Replit can be very funky.  When I set up my bots, I typically have to uninstall and reinstall Mineflayer and it's plugins multiple times for it to work.  This is all with the same code, so no guarantee it will work.  I **highly** recommend just using VS Code or pretty much any other online IDE than replit.  If it still doesn't work, you should try GitHub codespaces or vscode.dev.  Both are substantially better experiences

**Reply by Miloldr** [reply]
Also, it does work for once but I have to do kill 1 in shell. I could do os.system but now it doesn't do anything to shell, also if I wanted free 24/7, they made the free hosting not hosting
